README.md

```diff
@@ -111,6 +111,64 @@ When running on cluster node, you will also need to distribute this keytab, belo
+
+For other cases, SAC will create table related entities as Spark models.
+
+One exceptional case is HWC - for HWC source and/or sink, SAC will not create table related entities and always refer to Hive table entities via object id.
+
+Known Limitations (Design decision)
+====
+
+> SAC only supports SQL/DataFrame API (in other words, SAC doesn't support RDD).
+
+SAC relies on query listener to retrieve query and examine the impacts.
+
+> All "inputs" and "outputs" in multiple queries are accumulated into single "spark_process" entity when there're multple queries running in single Spark session.
+
+There're pros and cons regarding which is used for unit of "spark_process". Currently SAC uses "application ID" as unit - this approach could concentrate overall impacts on the single Spark process for who ran for when.
+
+But we also received other voices on this approach - lineage/relationship graph in "spark_process" are too complicated and confused and less meaningful. It's valid voice, we've filed #261 to investigate changing the unit of "spark_process" entity to query. It doesn't mean we will change it soon. It will be address only if we see the clear benefits on changing it.
+
+> Only part of inputs are tracked in Streaming query.
+
+This is from design choice on "trade-off": Kafka source supports subscribing with "pattern" and SAC cannot enumerate all matching existing topics, or even all possible topics (even it's possible it doesn't make sense, though).
+
+We found "executed plan" provides actual topics which each (micro) batch read and processed, but as a result, only inputs which participate on (micro) batch are being included as "inputs" in "spark_process" entity. 
+
+If your query are running long enough that it ingests data from all topics, it would have all topics in "spark_process" entity.
+
+> SAC doesn't support tracking changes on columns (Spark models).
+
+We found difficulty to make column entity be consistently "up-to-date" supposing multiple spark applications are running and each SAC in spark application are trying to update on Atlas. Due to the difficulty, we dropped supporting column tracking.
+
+We haven't found difficulty on table level, but once we collect difficulties on making table entity being consistent as well, we might also consider drop supporting DDL and blindly create the entities whenever they're referenced.
+
+This doesn't apply to Hive models, which central remote HMS takes care of DDLs and Hive Atlas Hook will take care of updates.
+
+> SAC doesn't track dropping tables (Spark models).
+
+"drop table" event from Spark only provides db and table name, which is NOT sufficient to create qualifierName - especially we separate two types of tables - spark and hive. 
+
+SAC depends on reading the Spark Catalog to get table information but Spark might already dropped the table when SAC notices the table is dropped, so that is NOT consistent. Due to this technical limitation, we have to drop this feature.
```

Comment:
```suggestion
SAC depends on reading the Spark Catalog to get table information but Spark will have already dropped the table when SAC notices the table is dropped so that will not work. Due to this technical limitation, we have to drop this feature.
```